scripts/generate-typings.js

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env node
+
+const fs = require('fs')
+const path = require('path')
+
+const PROTOCOLS = {
+    webdriver: require('../packages/webdriver/protocol/webdriver.json'),
+    appium: require('../packages/webdriver/protocol/appium.json'),
+    jsonwp: require('../packages/webdriver/protocol/jsonwp.json'),
+    mjsonwp: require('../packages/webdriver/protocol/mjsonwp.json')
+}
+
+const TEMPLATE_PATH = path.join(__dirname, 'templates', 'webdriver.tpl.d.ts')
+
+const lines = []
+for (const [protocolName, definition] of Object.entries(PROTOCOLS)) {
+    lines.push(`    // ${protocolName} types`)
+    lines.push('    interface Client<T> {')
+
+    for (const [, methods] of Object.entries(definition)) {
+        for (const [, description] of Object.entries(methods)) {
+            const { command, parameters = [], returns } = description
+            const params = parameters.map((p) => `${p.name}: ${p.type.toLowerCase()}`)
+            const returnValue = returns ? returns.type.toLowerCase() : 'undefined'
+            lines.push(`        ${command}(${params.join(', ')}): ${returnValue}`)
```

Comment:
While not 100% necessary, these lines should end with a semicolon

Reply:
This should also pull the ```required``` option, and if the parameter is not required then it should be set as optional